Issue#32: connect generated server to services

mykytam opened the pull request:
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @alexorlenko
- [x] @Arthurpanda
- [ ] @shiftko  
- [x] @mykytam 
- [ ] @omelchenkotimur 
- [ ] @Prosperro-de 
- [x] @reeztl 

### Second Level Review

- [ ] @DzigMS 
- [ ] @kh0ma 

## Summary of issue

We need to delegate the logic of generated services to our service beans so an implemented interface will be used in our application.

## Summary of change

Spring service is used instead of stub one

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

mykytam committed:
Override mustache templates and configure Jersey to Spring bridge

mykytam committed:
Fix date format in OpenAPI Specification file

mykytam committed:
Add custom service implementation

mykytam committed:
Issue#32
Fix pom versioning with Flatten plugin
Fix naming of generated packages
Fix using of servlet api

mykytam committed:
Implement package scanning for JerseyConfig

mykytam committed:
Delete redundant code

mykytam committed:
Update OpenApi spec

isvyry committed:
Issue#32 Connect generated server to services
Rearranged dependencies between parent pom and home-application pom;
Moved spring-boot-starter-jersey dependency into home-server-stub module.

isvyry committed:
Issue#32 Connect generated server to services
Delete spring boot maven version

mykytam committed:
Merge remote-tracking branch 'origin/dev' into Issue#32-Connect-generated-server-to-services

kh0ma commented on home-server-stub/pom.xml:
```diff
@@ -24,6 +24,10 @@
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-jersey</artifactId>
+        </dependency>
     </dependencies>
```
Please rename the current module&artifactId to `home-server-generator`.

mykytam replied:
Ok, noted

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/HomeApplication.java:
```diff
@@ -3,8 +3,8 @@
 
-@SpringBootApplication
-public class HomeApplication {
+@SpringBootApplication (scanBasePackages = "com.softserveinc.ita.homeproject")
```
Isn't `scanBasePackages` too wide?

mykytam replied:
Ok, noted

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/config/JerseyConfig.java:
```diff
@@ -0,0 +1,34 @@
+        // register endpoints
+        packages("com.softserveinc.ita.homeproject.api");
+        // register jackson for json
+        register(new ObjectMapperContextResolver(objectMapper));
```
Try to use com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider

mykytam replied:
Noted, we will try

mykytam committed:
Issue#32: connect generated server to services
Delete scanBasePackages
Rename module & artifactId to home-server-generator
Use JacksonJsonProvider instead of ObjectMapper

mykytam committed:
Issue#32: connect generated server to services
Bring back files from server-stub module

kh0ma commented on home-application/src/main/java/com/softserveinc/ita/homeproject/config/JerseyConfig.java:
```diff
@@ -0,0 +1,19 @@
+package com.softserveinc.ita.homeproject.config;
+
+import com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider;
```
Sorry for the inconvenience. I didn't notice (The same didn't you do). We have the generated one `import com.softserveinc.ita.homeproject.api.JacksonJsonProvider;` Please use it.

mykytam replied:
Sure, I will try to use it

kh0ma commented on home-server-generator/src/main/resources/mustaches/server/bootstrap.mustache:
```diff
@@ -0,0 +1,14 @@
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+
+public class Bootstrap {
```
Can't we exclude it from generating instead of adding an empty template?

mykytam replied:
Ok, noted

kh0ma commented on home-server-generator/src/main/resources/mustaches/server/pom.mustache:
```diff
@@ -127,11 +127,7 @@
       <version>${junit-version}</version>
       <scope>test</scope>
     </dependency>
-    <dependency>
```
Can't we just make it `provided` here?

mykytam replied:
Ok, noted

kh0ma commented on home-server-generator/src/main/resources/mustaches/server/pom.mustache:
```diff
@@ -206,6 +202,24 @@
+    <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-context</artifactId>
+        <version>2.0.5</version>
```
I wondered if `spring-beans` is enough here. But if it isn't the version of `spring-context` should correlate to `spring-beans`

mykytam replied:
Ok, I will try to use it

mykytam committed:
Issue#32: connect generated server to services
Delete scanBasePackages
Rename module & artifactId to home-server-generator
